### PR TITLE
Deprecate PackageResolver.packageRoot

### LINKED ISSUE #16: Deprecate APIs related to packageRoot

### PR DESCRIPTION
**Pull request by kevmoo**
Fixes https://github.com/dart-lang/package_resolver/issues/16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,17 +4,8 @@ isolate's package resolution strategy, but it may also be explicitly defined by
 the user—for example, you could create a resolver that represents the strategy
 used to compile a `.dart.js` file.
 
-[PackageResolver]: https://www.dartdocs.org/documentation/package_resolver/latest/package_resolver/PackageResolver-class.html
+The Dart VM provides resolving `package:` URIs using a a **package spec**.
```
Comment (nex3) [inline README.md:7]: "provides resolving" isn't grammatical.

File: lib/src/package_resolver.dart [hunk 2]
```diff
@@ -50,15 +47,12 @@ abstract class PackageResolver {
   ///
   /// Note that if this is a `data:` URI, it may not be safe to pass as a
   /// parameter to a Dart process due to length limits.
-  ///
-  /// Completes to `null` when using a [packageRoot] for resolution, or when no
-  /// package resolution is being used.
   Future<Uri> get packageConfigUri;
 
-  /// The base URL for resolving `package:` URLs.
+  /// **DEPRECATED** Will always return `null`.
```
Comment (nex3) [inline lib/src/package_resolver.dart:52]: This isn't true. A user can still construct a package resolver by hand that uses a package root to resolve URLs.

**Review by nex3 — CHANGES_REQUESTED** [review]

File: lib/src/package_resolver.dart (hunk 2 above)
Comment (kevmoo) [reply lib/src/package_resolver.dart:52]: I guess we could say that the default impl will return `null`...

**Comment by kevmoo** [comment]
The browser tests are all wonky now. Just going to punt on this for the moment...

**Pull request closed without merging**